drivers/spi/spi_nrfx_spim.c

```diff
@@ -571,21 +578,20 @@ static DEVICE_API(spi, spi_nrfx_driver_api) = {
 	.release = spi_nrfx_release,
 };
 
-#ifdef CONFIG_PM_DEVICE
 static int spim_nrfx_pm_action(const struct device *dev,
 			       enum pm_device_action action)
 {
-	int ret = 0;
+	int ret = -ENOTSUP;
 	struct spi_nrfx_data *dev_data = dev->data;
 	const struct spi_nrfx_config *dev_config = dev->config;
 
 	switch (action) {
+	case PM_DEVICE_ACTION_TURN_ON:
+		ret = 0;
+		break;
 	case PM_DEVICE_ACTION_RESUME:
 		ret = pinctrl_apply_state(dev_config->pcfg,
 					  PINCTRL_STATE_DEFAULT);
-		if (ret < 0) {
-			return ret;
-		}
```

Comment:
Why is this removed (also in `PM_DEVICE_ACTION_SUSPEND`)?

Reply:
it is aligned to how UARTE and PWM drivers behave in this manner - they ignore return code:
-  https://github.com/zephyrproject-rtos/zephyr/blob/fe0e2dbc6044ffa74b27bfcba335980bc5a6ae64/drivers/serial/uart_nrfx_uarte.c#L2193
- https://github.com/zephyrproject-rtos/zephyr/blob/fe0e2dbc6044ffa74b27bfcba335980bc5a6ae64/drivers/serial/uart_nrfx_uarte.c#L2265
- https://github.com/zephyrproject-rtos/zephyr/blob/fe0e2dbc6044ffa74b27bfcba335980bc5a6ae64/drivers/pwm/pwm_nrfx.c#L278
- https://github.com/zephyrproject-rtos/zephyr/blob/fe0e2dbc6044ffa74b27bfcba335980bc5a6ae64/drivers/pwm/pwm_nrfx.c#L317